integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala

```diff
@@ -171,39 +252,33 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
         .toList)
       if (driverRows.nonEmpty) {
         (Seq(
-          Row("ALL", "ALL", driverIndexSize, driverDatamapSize, allDictSize, "DRIVER"),
-          Row(currentDatabase,
-            "ALL",
-            driverdbIndexSize,
-            driverdbDatamapSize,
-            driverdbDictSize,
-            "DRIVER")
+          Row("TOTAL", driverIndexSize, driverDatamapSize, allDictSize, "DRIVER")
         ) ++ driverRows).collect {
-          case row if row.getLong(2) != 0L || row.getLong(3) != 0L || row.getLong(4) != 0L =>
-            Row(row(0), row(1), bytesToDisplaySize(row.getLong(2)),
-              bytesToDisplaySize(row.getLong(3)), bytesToDisplaySize(row.getLong(4)), "DRIVER")
+          case row if row.getLong(1) != 0L || row.getLong(2) != 0L || row.getLong(3) != 0L =>
+            Row(row(0), bytesToDisplaySize(row.getLong(1)),
+              bytesToDisplaySize(row.getLong(2)), bytesToDisplaySize(row.getLong(3)), "DRIVER")
         }
       } else {
-        makeEmptyCacheRows(currentDatabase)
+        makeEmptyCacheRows(currentDatabase, "DRIVER")
       }
     } else {
-      makeEmptyCacheRows(currentDatabase)
+      makeEmptyCacheRows(currentDatabase, "DRIVER")
     }
 
-    //      val (serverIndexSize, serverDataMapSize) = getAllIndexServerCacheSize
-    val indexDisplayRows = if (indexServerRows.nonEmpty) {
-      (Seq(
-        Row("ALL", "ALL", indexAllIndexSize, indexAllDatamapSize, indexAllDictSize, "INDEX SERVER"),
-        Row(currentDatabase,
-          "ALL",
-          indexdbIndexSize,
-          indexdbDatamapSize,
-          driverdbDictSize,
-          "INDEX SERVER")
-      ) ++ indexServerRows).collect {
-        case row if row.getLong(2) != 0L || row.getLong(3) != 0L || row.getLong(4) != 0L =>
-          Row(row.get(0), row.get(1), bytesToDisplaySize(row.getLong(2)),
-            bytesToDisplaySize(row.getLong(3)), bytesToDisplaySize(row.getLong(4)), "INDEX SERVER")
+    val indexDisplayRows = if (isDistributedPruningEnabled) {
+      if (indexServerRows.nonEmpty) {
+        (Seq(
+          Row("TOTAL", indexAllIndexSize, indexAllDatamapSize, indexAllDictSize, "INDEX SERVER")
+        ) ++ indexServerRows).collect {
+          case row if row.getLong(1) != 0L || row.getLong(2) != 0L || row.getLong(3) != 0L =>
+            Row(row.get(0),
+              bytesToDisplaySize(row.getLong(1)),
+              bytesToDisplaySize(row.getLong(2)),
+              bytesToDisplaySize(row.getLong(3)),
+              "INDEX SERVER")
+        }
+      } else {
+        makeEmptyCacheRows(currentDatabase, "INDEXSERVER")
```

Comment:
If there is nothing in the cache then no need to display anything.
We can remove this method itself and simplyfy the conditions.

Reply:
done